### PR TITLE
Assert correction for issue #17450

### PR DESCRIPTION
**Pull request by OresteMarquis**
The error message when FE_NedelecSZ are use to distribute dofs to the DoFHandler and then the VectorTools::interpolate function is called is now less cryptic. It prints : 

"The violated condition was: 
    fe_i.has_generalized_support_points()
Additional information: 
    The finite element does not have generalized support points. This is
    required for interpolation."
 
This was tested using the "error_c++_file.cc" presented in the initial issue (#17450).

### CODE REVIEW

**Review by masterleinad — APPROVED** [review]

**Comment by tamiko** [comment]
The spell checking failure is corrected on the main branch.

**Review by bangerth — APPROVED** [review]
Yes, this looks correct. Many thanks for writing the patch!

**Comment by bangerth** [comment]
The step-90 test now fails. 

@OresteMarquis  Can you talk to one of us about how to figure out what is going wrong there? I think that your patch is correct. We should figure out why step-90 uses a feature that can not work.

File: include/deal.II/numerics/vector_tools_interpolate.templates.h [hunk 1]
```diff
@@ -320,7 +321,14 @@ namespace VectorTools
       hp::QCollection<dim> support_quadrature;
       for (unsigned int fe_index = 0; fe_index < fe.size(); ++fe_index)
         {
-          const auto &fe_i   = fe[fe_index];
+          const auto &fe_i = fe[fe_index];
+          // If the finite element has no dofs, we can skip it
+          if (fe_i->dofs_per_cell == 0)
```
Comment (masterleinad) [inline include/deal.II/numerics/vector_tools_interpolate.templates.h:326]:
```suggestion
          if (fe_i.dofs_per_cell == 0)
```

**Review by tjhei — APPROVED** [review]

**Pull request merged**